src/OpenDiffix.Core/Anonymizer.fs

```diff
@@ -326,6 +326,24 @@ let countDistinct
     |> fst
     |> (Math.roundAwayFromZero >> int64 >> (+) safeCount >> AnonymizedResult.Ok)
 
+/// Returns the noisy count of a single AID set.
+let countDistinctAid
+  (anonParams: AnonymizationParams)
+  (anonContext: AnonymizationContext)
+  (numAids: int)
+  (aidSeed: Hash)
+  =
+  let noise =
+    [ anonContext.BucketSeed; aidSeed ]
+    |> generateNoise anonParams.Salt "noise" anonParams.LayerNoiseSD
+
+  let noisyCount = (float numAids + noise) |> Math.roundAwayFromZero |> int64
+
+  if noisyCount < anonParams.Suppression.LowThreshold then
+    AnonymizedResult.NotEnoughAIDVs
```

Comment:
`noisyCount` should be clamped to `LowThreshold` in this case. The `NotEnoughAIDVs` return value should be produced only when there aren't enough AID values for flattening, which we skip in this simplified version of the function.